src/carma-platform/guidance_scripts.py

```diff
@@ -163,5 +179,461 @@ def run_crosstrack_analysis(
 
     return (is_passed, stats, plt.gcf(), cross_tracks, timestamps)
 
+def process_actual_path(odom_data):
+    """Helper function that processes actual path data."""
+    return np.array([[point[0], point[1]] for point in odom_data])
+
+def process_planned_path(traj_plans):
+    """Helper function that processes planned path data with duplicate removal."""
+    planned_path = []
+    last_planned_point = None
+
+    for plan in traj_plans:
+        for point in plan:
+            if last_planned_point is None:
+                planned_path.append(point)
+                last_planned_point = point
+            else:
+                dist = np.linalg.norm(np.array(point) - np.array(last_planned_point))
+                if dist > 0.25:  # 0.25m threshold
+                    planned_path.append(point)
+                    last_planned_point = point
+                    break
+
+    return np.array(planned_path)
+
+def run_turn_accuracy_analysis(
```

Comment:
would you be able to also add the lane markings, so we can see the planned trajectory and actual pose within the lane?

Reply:
I tried doing this. Currently I am failing to make the visualization_marker array mesage type of the lanelet2 map to be deserialized by rosbag2_py library. It seems to be related to the fact that data is gathered by foxy system while the library is only available for humble.  So it may be little involved and I would suggest as future improvement or separate story

Reply:
From offline communication, got a help from Saina using a different library `mcap-ros2-support` to plot the data. The data had to be good as well it looks like because the one I had did not work with new library code either saying EndOfFille error.
Moreover, some approaches I did with rosbag2-py was still not working with good data. I didn't know about this library before and now I think whether if we should have used this from the beginning as it looks bestter without requiring the messages to be built...